src/main/java/FlowerStore/ItemManager.java

```diff
@@ -18,48 +31,85 @@ public void removeItemById(Integer id) {
-                    System.out.println("Sorted by itemName (ASC)");
-                } else if (order.equals(SortOrder.DESC)) {
-                    Collections.sort(this.items, (i1, i2) -> (i1.getItemName().compareTo(i2.getItemName())));
-                    Collections.reverse(this.items);
-                    System.out.println("Sorted by itemName (DESC)");
+        Comparator<Item> comparator = LIST_COMPARATOR.get(field);
+        if (order.equals(SortOrder.ASC)){
+            Collections.sort(this.items, comparator);
+        } else if (order.equals(SortOrder.DESC)){
+            Collections.sort(this.items, comparator);
+            Collections.reverse(this.items);
+        }
+    }
+
+    public List<BouquetFlower> compileBouquet(BouquetConfiguration configuration, List<Item> base){
+
+        // define configurations
+
+        List<BouquetFlower> baseFlowers = base.stream()
+                .filter(item -> item instanceof BouquetFlower)
+                .map(item -> (BouquetFlower) item)
+                .collect(Collectors.toList());
+        List<BouquetFlower> bouquet= new ArrayList<>();
+        String color = configuration.getColor();
+        int lowerBound = configuration.getPriceRange().getLowerBound();
+        int upperBound = configuration.getPriceRange().getUpperBound();
+        List<String> flower = configuration.getFlowers();
+        int count = configuration.getCount();
+        int flowerTypeCount = count / flower.size();
+        int priceTotal = 0;
+
+        // logic
+
+        for (String flowerType : flower) {
+            int i = 0;
+            for (BouquetFlower tempFlower : baseFlowers){
+                if (tempFlower.getItemName() == flowerType && tempFlower.getColor() == color && i < flowerTypeCount) {
+                        bouquet.add(tempFlower);
+                        priceTotal += tempFlower.getPrice();
+                        System.out.println(priceTotal);
+                        i++;
                 }
-                break;
-            case "price":
-                if (order.equals(SortOrder.ASC)) {
-                    Collections.sort(this.items, (i1, i2) -> i1.getPrice() - i2.getPrice());
-                    System.out.println("Sorted by price (ASC)");
-                } else if (order.equals(SortOrder.DESC)) {
-                    Collections.sort(this.items, (i1, i2) -> i1.getPrice() - i2.getPrice());
-                    Collections.reverse(this.items);
-                    System.out.println("Sorted by price (DESC)");
+            }
+        }
+        Comparator<Item> comparator = LIST_COMPARATOR.get("price");
+        Collections.sort(bouquet, comparator);
+        Collections.reverse(bouquet);
+        Collections.sort(baseFlowers, comparator);
+        Collections.reverse(baseFlowers);
+
+        int i = 0;
+        while (priceTotal > upperBound){
+            for (BouquetFlower bouquetFlower : bouquet){
+                for (BouquetFlower tempFlower : baseFlowers){
```

Comment:
It has complexity O(n^2) at least. I will try to think if it could be simplified

Reply:
I am not sure, if it's possible to write it better but I will try to optimize this once I have time